lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -260,25 +264,53 @@ xercesc::DOMNode* ParseXML::get_node(
                                       tag_name + " does not have the attribute " + att_name + "\n");
                         }
                     }
-                    // LIST element TODO refactor
-                    else if (&index == nullptr)
+                    // LIST element
+                    else if (index != nullptr)
                     {
-                        int32_t realIndex = index;
-                        // Transform index to real index
-                        if (index < 0)
+                        // Check value of index
+                        std::string in = *index;
+
+                        // Empty index
+                        if (in == "")
```

Comment:
```suggestion
                        if (index.empty())
```

std::string provides methods to check the string content.